### PR TITLE
Bug 1170090 - Intermittent failure in test_cost_control_data_alert_mobile.py at settings.reset_mobile_usage()

### PR DESCRIPTION
**Pull request by mwargers**

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c1803e729d48ac56718ab7e4b92d8535da951625

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f656ac40496442b1a42c847fa588cb7e3330652d

File: tests/python/gaia-ui-tests/gaiatest/apps/cost_control/regions/settings.py [hunk 1]
```diff
@@ -62,6 +64,26 @@ def select_when_use_is_above_unit_and_value(self, unit, value):
...
+
+        # Workaround for bug 1161088, where tapping on the button inside the app itself
+        # makes Marionette spew out NS_ERROR_NOT_INITIALIZED errors
+        x = confirm_reset_button.rect['x'] + confirm_reset_button.rect['width']//2
+        y = confirm_reset_button.rect['y'] + confirm_reset_button.rect['height']//2
+        self.marionette.switch_to_frame()
+        self.marionette.find_element(*self._screen_locator).tap(x=x, y=y)
+
+        # go back to iframe of usage app settings
+        Wait(self.marionette).until(lambda m: self.apps.displayed_app.src == self._src)
```
Comment (JohanLorenzo) [inline tests/python/gaia-ui-tests/gaiatest/apps/cost_control/regions/settings.py]: Why do we need to wait on the `_src`? Is it possible to wait on the app name?

File: tests/python/gaia-ui-tests/gaiatest/apps/cost_control/regions/settings.py [hunk 2]
```diff
@@ -62,6 +64,26 @@ def select_when_use_is_above_unit_and_value(self, unit, value):
...
+        self.apps.switch_to_displayed_app()
+        frame = Wait(self.marionette).until(expected.element_present(
```
Comment (JohanLorenzo) [inline tests/python/gaia-ui-tests/gaiatest/apps/cost_control/regions/settings.py]: We use the same logic in `__init__` to switch to the settings iFrame. I'd suggest to factorize it in a private function. It'll make `tap_confirm_reset` smaller.

File: tests/python/gaia-ui-tests/gaiatest/apps/cost_control/regions/settings.py (hunk 1 above)
Comment (mwargers) [inline tests/python/gaia-ui-tests/gaiatest/apps/cost_control/regions/settings.py]: I just saw this being used somewhere else in the code. I'll change this to be using the name.

File: tests/python/gaia-ui-tests/gaiatest/apps/cost_control/regions/settings.py (hunk 2 above)
Comment (mwargers) [inline tests/python/gaia-ui-tests/gaiatest/apps/cost_control/regions/settings.py]: Ok, will do!

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d4cefb470296bb8bbda4c4f3e83b88f3675bf31a

**Pull request merged**